tests/util/test_cli.py

```diff
@@ -0,0 +1,80 @@
+import json
+from inspect import signature
+from pathlib import Path
+
+import pytest
+import yaml
+
+from segmantic.util.cli import get_default_args, is_path, validate_args
+
+
+def function1(file_path: Path, arg_int: int, arg_float: float = -1.5):
+    pass
+
+
+def function2(arg_int: int, file_path: Path = None):
+    pass
+
+
+def test_get_default_args():
+    for fun in (function1, function2):
+        sig = signature(fun)
+
+        default_options = get_default_args(sig)
+        valid_args = validate_args(default_options, sig)
+
+        for k in sig.parameters:
+            if valid_args[k] is not None:
+                assert is_path(sig.parameters[k]) == isinstance(valid_args[k], Path)
+
+
+def test_validate_args():
+    args1 = {"file_path": "/path/file.txt", "arg_int": 10, "arg_float": 5.0}
```

Comment:
Nice!
Why dont' you parametrize these? 
When you have to test more variants it will be difficult to check whichone failed

https://docs.pytest.org/en/6.2.x/example/parametrize.html#paramexamples

Something like (please double check) ...
```python

params_data = {
 "args1_skip_default": ({"file_path": "/path/file.txt", "arg_int": 10}, False), 
  "args1_wrong_type": ({"file_path": 23, "arg_int": 10, "arg_float": 5}, True)
}

@pytest.mark.parametrize(
   "arguments,expected_failure", list(params_data.values()), ids=list(params_data.keys())
def test_validate_args(arguments, expected_failure):
   ...
```

Reply:
nice!